pkg/dataplane/reloadCredntials.go

```diff
@@ -0,0 +1,117 @@
+package dataplane
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/msi-dataplane/pkg/dataplane/swagger"
+	"github.com/fsnotify/fsnotify"
+	"github.com/go-logr/logr"
+)
+
+type reloadingCredential struct {
+	currentValue *azidentity.ClientCertificateCredential
+	cloud        string
+	lock         *sync.RWMutex
+	logger       *logr.Logger
+}
+
+type Option func(*reloadingCredential)
+
+func WithLogger(logger logr.Logger) Option {
+	return func(c *reloadingCredential) {
+		c.logger = &logger
+	}
+}
+
+func NewUserAssignedIdentityCredential(ctx context.Context, cloud string, credentialPath string, opts ...Option) (azcore.TokenCredential, error) {
```

Comment:
Let's add some GoDoc here - explain what `ctx` is for, why you might want to pass a logger, etc.

Reply:
Actually that reminds me - let's expose reload period for the ticker as an option.